webapp/channels/src/components/search_results/index.tsx

```diff
@@ -30,25 +30,15 @@ function makeMapStateToProps() {
     let results: Post[];
     let fileResults: FileSearchResultItem[];
     let files: FileSearchResultItem[] = [];
-    let posts: Post[];
     const addDateSeparatorsForSearchResults = makeAddDateSeparatorsForSearchResults();
 
     return function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
         const newResults = getSearchResults(state);
 
-        // Cache posts and channels
+        // Cache results
         if (newResults && newResults !== results) {
             results = newResults;
 
-            posts = [];
-            results.forEach((post) => {
-                if (!post) {
-                    return;
-                }
```

Comment:
This used to guard against `undefined` results, if we had actually used `posts` -- now we do it correctly in the selector.